classes/task/ReviewReminder.php

```diff
@@ -130,35 +61,108 @@ public function executeActions()
                 }
             }
 
-            if ($submission->getStatus() != PKPSubmission::STATUS_QUEUED) {
+            if ($submission->getData('status') != PKPSubmission::STATUS_QUEUED) {
                 continue;
             }
 
             // Fetch the context
-            if ($context == null || $context->getId() != $submission->getContextId()) {
+            if ($context == null || $context->getId() != $submission->getData('contextId')) {
                 unset($context);
-                $context = $contextDao->getById($submission->getContextId());
+                $context = $contextDao->getById($submission->getData('contextId'));
 
-                $inviteReminderDays = $context->getData('numDaysBeforeInviteReminder');
-                $submitReminderDays = $context->getData('numDaysBeforeSubmitReminder');
+                $numDaysBeforeReviewResponseReminderDue = $context->getData('numDaysBeforeReviewResponseReminderDue');
+                $numDaysAfterReviewResponseReminderDue = $context->getData('numDaysAfterReviewResponseReminderDue');
+
+                $numDaysBeforeReviewSubmitReminderDue = $context->getData('numDaysBeforeReviewSubmitReminderDue');
+                $numDaysAfterReviewSubmitReminderDue = $context->getData('numDaysAfterReviewSubmitReminderDue');
             }
 
             $mailable = null;
-            if ($submitReminderDays >= 1 && $reviewAssignment->getDateDue() != null) {
-                $checkDate = strtotime($reviewAssignment->getDateDue());
-                if (time() - $checkDate > 60 * 60 * 24 * $submitReminderDays) {
-                    $mailable = new ReviewRemindAuto($context, $submission, $reviewAssignment);
+            $currentDate = Carbon::today();
+
+            $dateResponseDue = Carbon::parse($reviewAssignment->getDateResponseDue())->startOfDay();
+            $dateDue = Carbon::parse($reviewAssignment->getDateDue())->startOfDay();
+
+            if ($reviewAssignment->getDateReminded() !== null) {
+                // we have a reminder sent previously
+
+                $dateReminded = Carbon::parse($reviewAssignment->getDateReminded())->startOfDay();
+
+                if ($reviewAssignment->getDateConfirmed() === null) {
+                    // review request has not been responded
+                    // previous reminder was a BEFORE REVIEW REQUEST RESPONSE reminder
+
+                    if ($numDaysAfterReviewResponseReminderDue &&
+                        $dateReminded->lt($dateResponseDue) &&
```

Comment:
I might not be reading this right, but I think this is a misunderstanding of the `date_reminded` column. I believe that gets set back to `null` when a review is confirmed -- see https://github.com/pkp/pkp-lib/blob/20b3178c9f1fb00724a31d73b98eff42131e9ca7/classes/submission/reviewer/ReviewerAction.php#L92..L97.